### PR TITLE
go: add head support for go 1.5

### LINKED ISSUE #41304: go --HEAD does not work because of changes in go 1.5
This is a ticket for discussions on how best to handle building go next. The --head option is a good place to get started on working on the kinks.

I just stepped through this process on my machine, so hopefully I can help. Go will look for an environment variable where it expects to find a working Go >=1.4 toolchain. It will then build Go 1.5 from this and install `go` `godoc` to `$GOPATH/bin`. This is the Go 1.5 binary and what needs to be provided to the end user. The `GOROOT_BOOTSTRAP` environment variable is used to locate Go.

### PR DESCRIPTION
**Pull request by drewwells**
Signed-off-by: Drew Wells <drew.wells00@gmail.com>

fixes #41304

Please see discussion in #41304 for comments on how to do this.

### CODE REVIEW

**Comment by mikemcquaid** [comment]
Might want to use a binary here rather than build-from-source (like we do with `ghc`).

File: Library/Formula/go.rb [hunk 1]
```diff
@@ -8,7 +8,7 @@ class Go < Formula
   sha1 "460caac03379f746c473814a65223397e9c9a2f6"
   version "1.4.2"
 
-  head "https://go.googlesource.com/go", :using => :git
+  head "https://github.com/golang/go", :using => :git
```
Comment (dunn) [inline Library/Formula/go.rb]: change the url to `https://github.com/golang/go.git` and you'll be able to drop the `:using => :git`

**Comment by drewwells** [comment]
We do not use the Go binaries provided by Google. Unless we are okay to change that decision.

If we did use provided binaries, what's the test matrix? 64bit, 32bit, lion-yosemite?

**Comment by mikemcquaid** [comment]
It'll run tests on 10.8 and above. This approach makes sense to me. Would be great if you could add a beta snapshot as a `devel do`, though, so we can throw CI at this. Thanks!

**Comment by drewwells** [comment]
Oh, I thought test ran against head. I'll add devel.

**Comment by drewwells** [comment]
There isn't a release branch for 1.5 yet. I've locked to the current master, so you can do tests. It'll need updating whenever the RC comes. 1.5 is supposed to be released in August, so RC should be coming soon...

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Comment by rasky** [comment]
After running `brew install go --devel`, I found out that the bootstrap compiler (a binary version of go 1.4.2, downloaded during formula installation) is kept in `/usr/local/Cellar/go/1.5beta2/libexec/gobootstrap`. That compiler shouldn't be needed after Go 1.5 is compiled. @drewwells

**Comment by drewwells** [comment]
Bootstrap tool chain wouldn't be required either. Feel free to pr cleaning
those up!

**Comment by DomT4** [comment]
^ #41912

**Comment by n054** [comment]
Just installed bootstrap yesterday. Will uninstall asap. Thank,nona

**Comment by drewwells** [comment]
Looks like PR was cherry picked instead of merged, deleted branch.

**Comment by bfontaine** [comment]
@drewwells Yep, that’s how we work on Homebrew.

**Comment by rasky** [comment]
Thanks! Since we're at it, go 1.5 doesn't require --with-cc-* anymore, since the whole runtime is recompiled transparently (lazy) as soon as a target architecture is selected (see http://dave.cheney.net/2015/03/03/cross-compilation-just-got-a-whole-lot-better-in-go-1-5).

**Pull request closed without merging**